File: src/main/java/com/github/kpavlov/jreactive8583/iso/MTI.java
package com.github.kpavlov.jreactive8583.iso;

import javax.annotation.Nonnull;

public class MTI {

    public static int mtiValue(@Nonnull ISO8583Version iso8583Version,
                               @Nonnull MessageClass messageClass,
                               @Nonnull MessageFunction messageFunction,
                               @Nonnull MessageOrigin messageOrigin
    ) {
        return iso8583Version.value() + messageClass.value() + messageFunction.value() + messageOrigin.value();
    }
}

File: src/main/kotlin/com/github/kpavlov/jreactive8583/iso/MTI.kt
@file:JvmName("MTI")

package com.github.kpavlov.jreactive8583.iso

public object MTI {

    @JvmStatic
    public fun mtiValue(
        iso8583Version: ISO8583Version,
        messageClass: MessageClass,
        messageFunction: MessageFunction,
        messageOrigin: MessageOrigin
    ): Int = (
        iso8583Version.value +
            messageClass.value +
            messageFunction.value +
            messageOrigin.value
        )
}
